File: front/projetoweb1/lib/models/banda_musica.dart
import 'musica.dart';
import 'banda.dart';

class BandaMusica {
  int? idBandaMusica;
  Banda? banda;
  Musica? musica;
  String? dataAdicao;

  BandaMusica({this.idBandaMusica, this.banda, this.musica, this.dataAdicao});

  factory BandaMusica.fromJson(Map<String, dynamic> json) {
    return BandaMusica(
      idBandaMusica: json['idBandaMusica'] as int?,
      banda: json['banda'] != null ? Banda.fromJson(json['banda']) : null,
      musica: json['musica'] != null ? Musica.fromJson(json['musica']) : null,
      dataAdicao: json['dataAdicao'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idBandaMusica': idBandaMusica ?? 0,
      'banda': banda?.toJson(),
      'musica': musica?.toJson(),
      'dataAdicao': dataAdicao,
    };
  }
}

File: back/projetowebI/src/main/java/com/projetowebI/models/BandaMusica.java
package com.projetowebI.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Data
@NoArgsConstructor
@Entity
public class BandaMusica {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idBandaMusica;

    @ManyToOne
    @JoinColumn(name = "id_banda")
    private Banda banda;

    @ManyToOne
    @JoinColumn(name = "id_musica")
    private Musica musica;

    private Date dataAdicao;

    public Integer getIdBandaMusica() {
        return idBandaMusica;
    }

    public void setIdBandaMusica(Integer idBandaMusica) {
        this.idBandaMusica = idBandaMusica;
    }

    public Banda getBanda() {
        return banda;
    }

    public void setBanda(Banda banda) {
        this.banda = banda;
    }

    public Musica getMusica() {
        return musica;
    }

    public void setMusica(Musica musica) {
        this.musica = musica;
    }

    public Date getDataAdicao() {
        return dataAdicao;
    }

    public void setDataAdicao(Date dataAdicao) {
        this.dataAdicao = dataAdicao;
    }
}